### DISCUSSION TITLE
Updating API namespace in a content model does not work

### DISCUSSION THREAD

**Discussion by coderreco** [swellstores/community#277] [Help] [answered]
### Context
I added a 'Select' content field to my 'Categories' content model. I am trying to use a select field to assign a general type to the category in order to group them together in my filters area on my front end.
![image](https://github.com/swellstores/community/assets/73268171/7470b201-0443-444f-a99a-c35b8932284a)

### Problem
When I edit the field I am trying to change the API namespace to 'Content' so that it is exposed to my front end, but every time I click 'Content', save and exit, then reopen the field editor the API namespace has reset to 'None'. I can confirm that the field does not show in the response when I call `swell.categories.list()`. If I view the 'Categories' model in the console the field does appear to show inside the 'Content' object. I seem to be experiencing this on a field I added to my products model as well.
![image](https://github.com/swellstores/community/assets/73268171/4a30188c-d05c-4a83-9a02-6311bf79cb88)

Despite a few bugs, I am really enjoying Swell as a software so far and enjoying learning the API. Looking forward to future updates to Swell to make better use of content models in the admin panel.

**Comment by coderreco** [comment]
### Update
I had previously deleted the unnecessary fields that are related to the origin theme in the admin panel but they still showed up in the console. I did a PUT request in the console in an attempt to remove those origin/storefront/theme related fields for good. It seemed like it worked, but I just checked the console again and they are still there.
![image](https://github.com/swellstores/community/assets/73268171/218b9e0c-7d04-4cd7-8275-5f0a913f03c8)

A side effect seemed to be that it reset each categories selection for the 'Type' field I am trying to expose to my front end. After going through and re-assigning them all, I can now see them on the front end. They still snow up as 'None' in the API namespace, though.

**Comment by sags95** [comment]
Thanks for reporting this @coderreco , I will take a look and file this with the team to investigate.

**Reply by coderreco** [reply]
Thanks @sags95,

Something else—I am instead trying creating a new content model called 'Category Group' where each item simply has a name and a boolean. This way I can create groups, toggle whether or not that group should be used as filters, then add it as a lookup field to my Category model and assign as necessary. 

For example, all of my categories that are brand names I would assign to the Category type called 'Brand', which would have a filterable value of 'true'.

It is connected and I can assign categories to groups, but when I query a category I only get the group ID. Can I expand that part of the query so that I can see the name and boolean value? I have seen similar questions here in the discussions. Is this something I need to add to my permissions? If so do you mind telling me what that permissions query is? I have tried to update permissions before and had trouble.

This is the response to a GET request on the model:
```
{
  "parent_id": "...",
  "name": "Latisse",
  "slug": "latisse",
  "top_id": "...",
  "date_created": "2023-04-22T04:53:52.725Z",
  "active": true,
  "date_updated": "2023-06-19T20:41:32.737Z",
  "sort": 0,
  "description": null,
  "images": null,
  "meta_description": null,
  "meta_title": null,
  "sorting": null,
  "type": "brand",
  "undefined": null,
  "content": {
    "group_id": "6490b2bd12a7620014e51228"
  },
  "id": "64436860c248cd00129fcaa9"
}
```

**Reply by sags95** [reply, marked as answer]
Yes many of the posts here are referring to "opening permissions" which is referring to this guide [here](https://developers.swell.is/guides/opening-content-permissions). This refers to the Frontend API specifically, as by default it only returns certain fields (some of which have different levels of permissions). So opening these permissions allows additional fields to be returned and modified for write-access.

In your case above, since it's related to the Backend API (GET), there is no need to modify the permissions since all fields are accessible. One thing that you can try is by passing in a `include` parameter to get the other values https://developers.swell.is/backend-api/querying/including

**Comment by logeshswell** [comment]
> Updating API namespace in a content model does not work

@coderreco I can confirm that it is a dashboard UI bug but it doesn't have anything to do wrongly with the functionality.
I will create a bug ticket internally for faster resolution.

> I can confirm that the field does not show in the response when I call swell.categories.list()

The field will be returned only if you set some value to it.

**Reply by logeshswell** [reply]
This bug is fixed now. Please check and confirm.
